Kudu.Services/Diagnostics/ProcessController.cs

```diff
@@ -274,6 +276,48 @@ public HttpResponseMessage MiniDump(int id, int dumpType = 0, string format = nu
             }
         }
 
+        [HttpPost]
+        public async Task<HttpResponseMessage> TakeMemoryDumpOnCrash(int id)
+        {
+            using (_tracer.Step("ProcessController.TakeMemoryDumpOnTerminate"))
+            {
+                try
+                {
+                    var process = GetProcessById(id);
+                    var externalCommandFactory = new ExternalCommandFactory(_environment, _settings, _environment.RootPath);
+                    var exe = externalCommandFactory.BuildExternalCommandExecutable(_environment.CrashDumpsPath, _environment.WebRootPath, NullLogger.Instance);
+                    var outputStream = new MemoryStream();
+                    var errorStream = new MemoryStream();
+
+                    var runTask = Task.Run(() => exe.ExecuteAsync(_tracer, $"{ResolveProcDumpPath()} -accepteula -t {id}", outputStream, errorStream, idleManager: new IdleManager(TimeSpan.MaxValue, _tracer)));
+
+                    // procDump can fail to attach to a process if there is another instance of procDump (or any debugger) that's already attached to the same process
+                    await Task.Delay(TimeSpan.FromSeconds(1));
```

Comment:
It was arbitrary since the whole workflow is asynchronous. The way `procdump.exe -t {pid}` works is that it waits for ever until the process `{pid}` terminates, and it takes a dump then. However it can fail when you first launch it due to things like access denied, won't happen in the sandbox, or "debugger already attached", which could happen if the process you're trying to monitor is already being debugged by any other debugger. I kind of just picked 1 second to wait and see if it terminates within that then report the error, otherwise I'll assume everything is fine.

Now that I think about it, I think we may not need to handle any failure case at all here. Just let the tracer write any errors to kudu trace logs and always report success if the process just starts without any unexpected errors. What do you think?